### DISCUSSION TITLE
Invoke-FalconRTR -OfflineQueue

### DISCUSSION THREAD

**Discussion by FortifyUnited** [CrowdStrike/psfalcon#234] [Q&A] [answered]
I am working with PSFalcon and have been running scripts from there. I have a question pertaining to the -OfflineQueue $true. Does anyone know how long they stay in queue?

**Comment by bk-cs** [comment, marked as answer]
The offline queue lasts for 7 days. If the host targeted by the session does not come online within those 7 days, the session expires.

You can reset the 7 days by appending new commands to an existing session (which happens automatically if you `-QueueOffline $true` against a host that has an existing queued session), or you can let it expire and re-issue the original commands (i.e. create a new queued session).

**Reply by FortifyUnited** [reply]
By chance where is your source?

**Reply by bk-cs** [reply]
Source of my comments? I am a CrowdStrike employee and familiar with how the RTR APIs work.

**Reply by FortifyUnited** [reply]
Oh that is fantastic. Thank you for the insight. Your information is much appreciated.
